_kbarticles/2025-03-27-monitoring-app-metrics-opensearch.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: "Monitoring your application's health using OpenSearch"
+date: March 27, 2025
+excerpt: How to monitor the CPU and memory metrics for your application on Cloud.gov using OpenSearch
+---
+
+One of the benefits of hosting your applications on Cloud.gov is that it offers [performance metrics for
+monitoring the health of your application](https://docs.cloudfoundry.org/loggregator/container-metrics.html).
+
+The most important metrics for monitoring your application's health are the memory and CPU metrics, which can help you identify:
+
+- If your application has enough memory provisioned per instance
+- If your application is experiencing CPU spikes and associated performance issues
+- If you need [to horizontally scale your application instances]({{ site.baseurl }}{% link _kbarticles/2023-08-07-advantages-of-multiple-application-instances.md %})
+
```

Comment:
Add (if Platform engineers concur):

- If your application needs more memory allocation because it's exceeding its CPU entitlement

Reply:
The reason why I didn't specify CPU entitlement is because applications can exceed CPU entitlement and not experience performance issues. Typically the performance issues only occur when they are throttled while borrowing CPU because their neighbors on the cell are also using up their CPU and taking away the amount that was borrowed while under load.